test/e2e/external_oidc_test.go

```diff
@@ -187,263 +439,311 @@ func newTestClient(t *testing.T) (*testClient, error) {
+		{schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"}, "openshift-authentication", "v4-0-config-system-session"},
+		{schema.GroupVersionResource{Group: "", Version: "v1", Resource: "services"}, "openshift-authentication", "oauth-openshift"},
+		{schema.GroupVersionResource{Group: "apiregistration.k8s.io", Version: "v1", Resource: "apiservices"}, "", "v1.oauth.openshift.io"},
+		{schema.GroupVersionResource{Group: "apiregistration.k8s.io", Version: "v1", Resource: "apiservices"}, "", "v1.user.openshift.io"},
+		{schema.GroupVersionResource{Group: "oauth.openshift.io", Version: "v1", Resource: "oauthclients"}, "", "openshift-browser-client"},
+		{schema.GroupVersionResource{Group: "oauth.openshift.io", Version: "v1", Resource: "oauthclients"}, "", "openshift-challenging-client"},
+		{schema.GroupVersionResource{Group: "oauth.openshift.io", Version: "v1", Resource: "oauthclients"}, "", "openshift-cli-client"},
+	} {
+		_, err := dynamicClient.Resource(obj.gvr).Namespace(obj.namespace).Get(ctx, obj.name, metav1.GetOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			return false, fmt.Errorf("unexpected error while getting resource %s/%s: %v", obj.namespace, obj.name, err)
+		} else if requireMissing != errors.IsNotFound(err) {
+			return false, fmt.Errorf("resource %s/%s wanted missing: %v; got: %v", obj.namespace, obj.name, requireMissing, !errors.IsNotFound((err)))
+		}
 	}
 
-	_, err = tc.kubeClient.CoreV1().ConfigMaps(kasNamespace).Create(ctx, oidcServingCA, metav1.CreateOptions{})
-	cleanups = append(cleanups, func() {
-		err := tc.kubeClient.CoreV1().ConfigMaps(kasNamespace).Delete(ctx, oidcServingCA.Name, metav1.DeleteOptions{})
+	// routes
+	for _, obj := range []struct{ namespace, name string }{
+		{"openshift-authentication", "oauth-openshift"},
+	} {
+		_, err := tc.routeClient.RouteV1().Routes(obj.namespace).Get(ctx, obj.name, metav1.GetOptions{})
 		if err != nil && !errors.IsNotFound(err) {
-			tc.t.Fatalf("cleanup failed for secret '%s/%s': %v", oidcServingCA.Namespace, oidcServingCA.Name, err)
+			return false, fmt.Errorf("unexpected error while getting route %s/%s: %v", obj.namespace, obj.name, err)
+		} else if requireMissing != errors.IsNotFound(err) {
+			return false, fmt.Errorf("route %s/%s wanted missing: %v; got: %v", obj.namespace, obj.name, requireMissing, !errors.IsNotFound((err)))
+		}
+
+		// ingress status
+		ingress, err := tc.configClient.ConfigV1().Ingresses().Get(ctx, "cluster", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		found := false
+		for _, route := range ingress.Status.ComponentRoutes {
+			if route.Name == obj.name && route.Namespace == obj.namespace {
+				found = true
+				break
+			}
+		}
+
+		if requireMissing != !found {
+			return false, fmt.Errorf("route %s wanted missing: %v; got: %v", obj, requireMissing, !found)
```

Comment:
Make it explicit in the error message that it should be missing or not from the cluster ingress status?